packages/web-components/src/toggle-button/toggle-button.ts

```diff
@@ -1,108 +1,74 @@
-    }
+  /**
+   * Updates the pressed state when the `pressed` property changes.
+   *
+   * @internal
+   */
+  protected pressedChanged(): void {
+    this.setPressedState();
   }
 
   /**
-   * The checked state of the control.
+   * Indicates the indeterminate state of the control. This property takes precedence over `pressed`.
    *
    * @public
+   * @remarks
+   * HTML Attribute: `indeterminate`
    */
-  @observable
-  public checked: boolean = false;
-  protected checkedChanged(prev: boolean | undefined, next: boolean): void {
-    if (!this.$fastController.isConnected) {
-      return;
-    }
-
-    if (!this.dirtyChecked) {
-      this.dirtyChecked = true;
-    }
-
-    this.currentChecked = this.checked;
-
-    ((this as unknown) as HTMLElement).setAttribute('aria-pressed', `${this.currentChecked}`);
-
-    if (prev !== undefined) {
-      this.$emit('change');
-    }
-  }
+  @attr({ mode: 'boolean' })
+  public indeterminate?: boolean;
 
   /**
-   * The current checkedness of the element. This property serves as a mechanism
-   * to set the `checked` property through both property assignment and the
-   * .setAttribute() method. This is useful for setting the field's checkedness
-   * in UI libraries that bind data through the .setAttribute() API
-   * and don't support IDL attribute binding.
+   * Updates the pressed state when the `indeterminate` property changes.
+   *
+   * @param previous - the previous indeterminate state
+   * @param next - the current indeterminate state
+   * @internal
    */
-  @attr({ attribute: 'current-checked', mode: 'boolean' })
-  public currentChecked: boolean = false;
-  public currentCheckedChanged(prev: boolean | undefined, next: boolean) {
-    this.checked = this.currentChecked;
+  protected indeterminateChanged(): void {
+    this.setPressedState();
   }
 
-  constructor() {
-    super();
-
-    // Re-initialize dirtyChecked because initialization of other values
-    // causes it to become true
-    this.dirtyChecked = false;
+  /**
+   * Toggles the pressed state of the button.
+   *
+   * @override
+   */
+  protected press(): void {
+    this.pressed = !this.pressed;
   }
 
-  public connectedCallback() {
+  connectedCallback(): void {
     super.connectedCallback();
-
-    ((this as unknown) as HTMLElement).addEventListener('click', this.handleToggleButtonClick);
-  }
-
-  public disconnectedCallback() {
-    super.disconnectedCallback();
-
-    ((this as unknown) as HTMLElement).removeEventListener('click', this.handleToggleButtonClick);
+    this.setPressedState();
   }
 
   /**
+   * Sets the `aria-pressed` attribute based on the `pressed` and `indeterminate` properties.
+   *
    * @internal
    */
-  protected handleToggleButtonClick = (e: MouseEvent): void => {
-    if (!this.disabled && !this.disabledFocusable) {
-      this.checked = !this.checked;
-    }
-  };
+  private setPressedState(): void {
+    const ariaPressed = `${this.indeterminate ? 'mixed' : !!this.pressed}`;
```

Comment:
Do toggle buttons support mixed? I think that would simply be for `checkbox`. I don't think `switch` would be mixed either. https://www.w3.org/WAI/ARIA/apg/patterns/button/

I'm not sure how you'd have a mixed state for on/off. I know that it's supported in "pressed" but I am not sure it makes sense in the pattern.

Reply:
From MDN:

> Adding `aria-pressed` to an element with a role of [`button`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) turns the button into a toggle button. The `aria-pressed` attribute is only relevant for toggle buttons. It represents the button's current "pressed" state.
> 
> The value is "tri-state", meaning the value can be set to `true`, `false`, `mixed`, or `undefined` values. In the case of `aria-pressed`, like most tri-state value types, the default value is `undefined`.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-pressed

I can remove the `indeterminate`/`mixed` state if it isn't relevant for this particular component - I don't think we have any stylistic differences for a mixed-state toggle button.

Reply:
Let's leave it for now given this documentation - I'm not sure it infers that there should be a visual difference here, rather it pertains to the value of what this represents. Good to leave for now.